src/main/java/org/candlepin/async/tasks/CloudAccountOrgSetupJob.java

```diff
@@ -0,0 +1,227 @@
+                JobConstraints.uniqueByArguments(CLOUD_ACCOUNT_ID, OFFERING_ID)).setRetryCount(5);
+        }
+
+        /**
+         * Sets the cloud account ID for this job
+         *
+         * @param cloudAccountId
+         *     the cloud account ID to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudAccountId(String cloudAccountId) {
+            if (cloudAccountId == null || cloudAccountId.isEmpty()) {
+                throw new IllegalArgumentException("cloudAccountId is null or empty");
+            }
+
+            this.setJobArgument(CLOUD_ACCOUNT_ID, cloudAccountId);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud offering ID for this job
+         *
+         * @param cloudOfferingId
+         *     the cloud offering ID to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudOfferingId(String cloudOfferingId) {
+            if (cloudOfferingId == null || cloudOfferingId.isEmpty()) {
+                throw new IllegalArgumentException("cloudOfferingId is null or empty");
+            }
+
+            this.setJobArgument(OFFERING_ID, cloudOfferingId);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud provider for this job
+         *
+         * @param cloudProviderShortName
+         *     shortcut of the cloud provider
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudProvider(CloudProvider cloudProviderShortName) {
+            if (cloudProviderShortName == null) {
+                throw new IllegalArgumentException("cloudProviderShortName is null");
+            }
+
+            this.setJobArgument(CLOUD_PROVIDER, cloudProviderShortName);
+
+            return this;
+        }
+
+        /**
+         * Sets the owner key for this job
+         *
+         * @param ownerKey
+         *     the owner key to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setOwnerKey(String ownerKey) {
+            if (ownerKey.isEmpty()) {
+                throw new IllegalArgumentException("ownerKey is empty");
+            }
+
+            this.setJobArgument(OWNER_KEY, ownerKey);
+
+            return this;
+        }
+
+        @Override
+        public void validate() throws JobConfigValidationException {
+            super.validate();
+
+            try {
+                JobArguments arguments = this.getJobArguments();
+
+                String accountId = arguments.getAsString(CLOUD_ACCOUNT_ID);
+                String offeringId = arguments.getAsString(OFFERING_ID);
+                String cloudProviderShortName = arguments.getAsString(CLOUD_PROVIDER);
+
+                if (accountId == null || accountId.isEmpty()) {
```

Comment:
can we change all the `isEmpty` method calls into `isBlank`, because that also covers the case the string is only whitespace. Not just in this method, but everywhere really

Reply:
done